subsys/zbus/zbus.c

```diff
@@ -371,6 +371,11 @@ int zbus_chan_pub(const struct zbus_channel *chan, const void *msg, k_timeout_t
 		return err;
 	}
 
+#if defined(CONFIG_ZBUS_CHANNEL_PUBLISH_STATS)
+	chan->data->publish_timestamp = k_uptime_ticks();
+	chan->data->publish_count += 1;
+#endif /* CONFIG_ZBUS_CHANNEL_PUBLISH_STATS */
```

Comment:
If this is pushed after VDED, subscribers and observers will see inconsistent state depending on what functions are called.
If they are run because of `zbus_chan_pub`, they will see statistics from the previous publish.
If they are run because of `zbus_chan_notify`, they will see statistics from the current publish.
If the values are different with no way to determine which case you are in, they are useless.

`zbus_chan_notify` is about forcing channel notifications, not publishing.
I see the value of the statistics in that you can determine when the channel data was generated, not the last time someone decided to call `zbus_chan_notify`.
If observers want to know how often they are called, they can trivially keep track of that information themselves.